### PR TITLE
Diagnoses Consul Direct Access Check

### PR DESCRIPTION
**Pull request by HridoyRoy**
The logic checks if address is not localhost for any storage or SR configuration stanza. Warnings are moved before errors so they can be relayed even when a particular error occurs.

### CODE REVIEW

File: vault/diagnose/storage_checks.go [hunk 1]
```diff
@@ -80,3 +81,14 @@ func StorageEndToEndLatencyCheck(ctx context.Context, b physical.Backend) error
 	}
 	return nil
 }
+
+// ConsulDirectAccess verifies that consul is connecting to local agent,
+// versus directly to a remote server. We can only assume that the local address
+// is a server, not a client.
+func ConsulDirectAccess(config map[string]string) string {
+	serviceRegistrationAddr := config["address"]
```
Comment (swayne275) [inline vault/diagnose/storage_checks.go:89]: do we want to handle the case where `address` isn't in `config`?

File: vault/diagnose/storage_checks.go [hunk 2]
```diff
@@ -80,3 +81,14 @@ func StorageEndToEndLatencyCheck(ctx context.Context, b physical.Backend) error
...
+	if !strings.Contains(serviceRegistrationAddr, "localhost") && !strings.Contains(serviceRegistrationAddr, "127.0.0.1") {
+		return "consul storage does not connect to local agent, but directly to server"
```
Comment (swayne275) [inline vault/diagnose/storage_checks.go:91]: does it make sense to `const` this since we check for it in tests?

**Review by swayne275 — APPROVED** [review]
few comments but it looks good

File: vault/diagnose/storage_checks.go (hunk 1 above)
Comment (HridoyRoy) [reply vault/diagnose/storage_checks.go:89]: Yes, good point!

File: vault/diagnose/storage_checks.go (hunk 2 above)
Comment (HridoyRoy) [reply vault/diagnose/storage_checks.go:91]: Absolutely.

**Pull request merged**